src/fields2cover/route_planning/route_planner_base.cpp

```diff
@@ -18,47 +18,66 @@ namespace f2c::rp {
 
 namespace ortools = operations_research;
 
-F2CRoute RoutePlannerBase::genRoute(
-    const F2CCells& cells, const F2CSwathsByCells& swaths,
-    bool show_log, double d_tol, bool redirect_swaths) {
+F2CRoute RoutePlannerBase::genRoute(const F2CCells& cells,
+    const F2CSwathsByCells& swaths, bool show_log, double d_tol,
+    bool redirect_swaths) {
   F2CGraph2D shortest_graph = createShortestGraph(cells, swaths, d_tol);
 
   F2CGraph2D cov_graph = createCoverageGraph(
       cells, swaths, shortest_graph, d_tol, redirect_swaths);
 
   std::vector<int64_t> v_route = computeBestRoute(cov_graph, show_log);
-  return transformSolutionToRoute(
-      v_route, swaths, cov_graph, shortest_graph);
+  return transformSolutionToRoute(v_route, swaths, cov_graph, shortest_graph);
+}
+
+F2CRoute RoutePlannerBase::genShortestRoute(const F2CCells& cells,
+    const F2CSwathsByCells& swaths, const F2CPoint& start, const F2CPoint& end,
+    double d_tol) {
+  F2CRoute route;
+  f2c::rp::RoutePlannerBase route_planner;
+  auto g = route_planner.createShortestGraph(cells, swaths, d_tol);
```

Comment:
Creating this graph is computationally expensive. Receive one shortest graph as a parameter instead

Reply:
You can create it outside this function, and receive it as an argument of the function.
This function is quite useful, but I'm concerned it will be used iteratively many times, requiring to create the same graph many times, which is expensive. 

As an additional note: If you want to modify the `swaths` to create different graphs, it's faster to save an original graph with all the edges, copy the graph and remove the edges you don't want from that copy.

Reply:
I want to expose it as a top level function, so I'm still not sure how I would create the graph before calling it

Reply:
I guess it's fine then